Rear Rider/Rear Rider/Auth/SignUpView.swift

```diff
@@ -0,0 +1,103 @@
+//
+//  SignUpView.swift
+//  timerr
+//
+//  Created by Robert Paul Sutton on 1/27/22.
+//
+
+import SwiftUI
+import Firebase
+
+struct SignUpView: View {
+    
+    @EnvironmentObject var viewRouter: ViewRouter
+    @EnvironmentObject var firestoreModel: FirestoreModel
+    @EnvironmentObject var auth: AuthModel
+    @State var email = ""
+    @State var password = ""
+    @State var passwordConfirmation = ""
+    @State var signUpErrorMessage = ""
+    
+    var body: some View {
+        VStack(spacing: 15) {
+            Text("Rear Rider").font(.title)
+            SignUpCredentialFields(email: $email, password: $password, passwordConfirmation: $passwordConfirmation)
+            Button(action: {
+                signUp()
+            }) {
+                Text("Sign Up")
+                    .bold()
+                    .frame(width: 360, height: 50)
+                    .background(.blue)
+                    .foregroundColor(.white)
+                    .cornerRadius(10)
+                    .opacity(!auth.authLoading && !email.isEmpty && !password.isEmpty && !passwordConfirmation.isEmpty && password == passwordConfirmation ? 0.9 : 0.4)
+            }
+            .disabled(!auth.authLoading && !email.isEmpty && !password.isEmpty && !passwordConfirmation.isEmpty && password == passwordConfirmation ? false : true)
```

Comment:
same here with using OR and flipping the ternary:

```suggestion
                    .opacity(auth.authLoading || email.isEmpty || password.isEmpty || passwordConfirmation.isEmpty || password != passwordConfirmation ? 0.4 : 0.9)
            }
            .disabled(auth.authLoading || email.isEmpty || password.isEmpty || passwordConfirmation.isEmpty || password != passwordConfirmation ? true : false)
```